plugins/query-insights/src/main/java/org/opensearch/plugin/insights/rules/transport/top_queries/TransportTopQueriesAction.java

```diff
@@ -81,17 +78,18 @@ protected TopQueriesResponse newResponse(
         final List<TopQueries> responses,
         final List<FailedNodeException> failures
     ) {
-        if (topQueriesRequest.getMetricType() == MetricType.LATENCY) {
-            return new TopQueriesResponse(
-                clusterService.getClusterName(),
-                responses,
-                failures,
-                clusterService.getClusterSettings().get(QueryInsightsSettings.TOP_N_LATENCY_QUERIES_SIZE),
-                MetricType.LATENCY
-            );
-        } else {
-            throw new OpenSearchException(String.format(Locale.ROOT, "invalid metric type %s", topQueriesRequest.getMetricType()));
+        int size;
+        switch (topQueriesRequest.getMetricType()) {
```

Comment:
Don't need to handle invalid metric type for JVM here right ?

Reply:
No, that's already handled on the REST level.